data-generation/src/main/scala/LogTSV.scala

```diff
@@ -0,0 +1,176 @@
+import Action.{CREATE, DELETE, UPDATE}
+import LTSV.Attributes
+import com.github.javafaker.Faker
+
+import java.time.{Instant, LocalDateTime, ZoneOffset}
+import scala.util.Try
+import scala.util.Random.nextInt
+import scala.io.Source
+
+/**
+ * This is apparently the best way to make enums in Scala2
+ */
+sealed abstract class Action
+object Action{
+  final case object CREATE extends Action
+  final case object UPDATE extends Action
+  final case object DELETE extends Action
+}
+
+/** LogTSV
+ *
+ * Everything is built around this case class
+ * @param timestamp When did the log_entry happen?
+ * @param action Type of action
+ * @param objectId Id of the object
+ * @param attributes List of (Key,Value) attributes relevant to the entry
+ */
+case class LogTSV(
+                 timestamp : Instant,
+                 action : Action,
+                 objectId: Long,
+                 attributes: Attributes
+                 )
+
+object LTSV {
+
+  /** Random LogTSV instance
+   *
+   * Convenient for testing
+   * @return An instance of LogTSV
+   */
+  def randomLTSV():LogTSV = {
+    val actions = List(CREATE, UPDATE, DELETE)
+    val faker = new Faker()
+    LogTSV(
+      timestamp = Instant.now(),
+      action = actions(nextInt(actions.size)),
+      objectId = nextInt(10000),
+      attributes = List(("champion",faker.leagueOfLegends().champion()),
+        ("friends", faker.friends().character()))
+    )
+
+
+
+  }
+  // Type alias
+  type Attributes = List[(String, String)]
+
+  /** Serialize a single LogTSV
+   *
+   * @param log_entry The log entry
+   * @return String representation of the LogTSV
+   */
+  def serializeLTSV(log_entry:LogTSV ): String ={
+    val timestamp = LocalDateTime.ofInstant(log_entry.timestamp, ZoneOffset.UTC).toString +"Z"
+    val action = log_entry.action match {
+      case CREATE => "CREATE"
+      case UPDATE => "UPDATE"
+      case DELETE => "DELETE"
+    }
+    val objectId = log_entry.objectId.toString
+    val attributes = serializeAttributes(log_entry.attributes)
+
+    List(timestamp, action, objectId, attributes)
+      .mkString("\t")
+  }
```

Comment:
Some formatting + casing conventions
```suggestion
  def serializeLTSV(logEntry: LogTSV): String = {
    val timestamp = LocalDateTime.ofInstant(logEntry.timestamp, ZoneOffset.UTC).toString + "Z"
    val action = logEntry.action match {
      case CREATE => "CREATE"
      case UPDATE => "UPDATE"
      case DELETE => "DELETE"
    }
    val objectId = logEntry.objectId.toString
    val attributes = serializeAttributes(logEntry.attributes)

    List(timestamp, action, objectId, attributes)
      .mkString("\t")
  }
```

Reply:
Nice, usikker på om jeg kjørte linteren. Skal sjekke ut det også